CONTRIBUTING.md

```diff
@@ -107,10 +107,30 @@ This section shows you how to send a pull request. The main staff are always mon
 
 This section shows you how to write commit messages in Clay. Follow these guidelines to help us maintain order and make it easier to locate your changes.
 
-#### Before You Make a Commit Message
+Each commit message consists of a header, a body and a footer. The header has a special format that includes a type, a scope and a subject:
```

Comment:
What is the body and header portion?

Reply:
I think this more or less reflects the structure of the header, footer and body.

```markdown
<type>[optional scope]: <description> // <-- Header

[optional body]

[optional footer]
```

I don't know if that's what you wanted to know.

Reply:
I think I would just remove the portion about the header/body/footer and only specifically mention the type, scope, and subject. Perhaps something like

```
Each commit message should include a typecheck, scope, and subject. It should be formatted as follows:
```

Reply:
It's also possible that all of this could be automatically enforced [via bot](https://github.com/probot/semantic-pull-requests).

Reply:
@bryceosterhaus, in the https://www.conventionalcommits.org/ specification, they say more about it, but we can just go with the header.

Oh I had forgotten about the @wincent bot, we can add it here. What do you think?